default.nix

```diff
@@ -2,15 +2,23 @@
 , compiler ? "default"
 }:
 let
-  pkgs = import nixpkgs {
-    config.allowBroken = true;
-    overlays = [(import ./waargonaut-deps.nix)];
+  overlay = self: super: 
+    let
+      baseHaskellPackages = if compiler == "default"
```

Comment:
Why this isn't a function in nixpkgs I'll never know

Reply:
Probably because we haven't put it there...